prime-router/src/main/kotlin/common/HttpClientUtils.kt

```diff
@@ -25,13 +25,41 @@ import kotlinx.serialization.json.Json
 
 class HttpClientUtils {
     companion object {
+
+        /**
+         * The @Volatile annotation is needed to ensure that the instance property is updated atomically.
+         * This prevents other threads from creating more instances and breaking the singleton pattern.
+         */
+
+        @Volatile
+        private var httpClient: HttpClient? = null
+
+        @Volatile
+        private var httpClientWithAuth: HttpClient? = null
+
+        // the httpClient object does not provide a direct means of inspecting its configuration and we need to know
+        // what auth token was supplied to the existing client so we can make a determination as to whether or not to
+        // return the existing object or return a new one. While the raw token is already in memory as a member of the
+        // httpClientWithAuth object and the risk is very low we're using the hash for defense-in-depth purposes.
+        @Volatile
+        private var accessTokenHash: Int = 0
+
         /**
          * timeout for http calls
          */
         private const val TIMEOUT = 50_000
         const val REQUEST_TIMEOUT_MILLIS: Long = 130000 // need to be public to be used by inline
         const val SETTINGS_REQUEST_TIMEOUT_MILLIS = 30000
 
+        /**
+         * resets client and token hash to default
+         */
+        private fun reset() {
```

Comment:
Is this only here for testing?

Reply:
yup. this is a static/companion class and this was the simplest and most grokkable way I could think of to ensure the tests had a "fresh" copy of those vars for each test run.

Reply:
I would just make this a public method and rename it to `destroy` or `deinit`. It's a functionality that could be useful in the code and cleans up access in the tests. And maybe add a quick unit tests that makes sure destroy functions as expected and doesn't break the class somehow.

Reply:
I don't see a reason why this can't be public? This seems like reasonable functionality a potential client (calling code) would use?

Reply:
we can easily make it public in future if some other prod code needs it. Until we actually need that functionality though I'd rather keep it private and prevent future shenanigans. The point of this ticket was to reduce the number of times the client gets re-created vs re-used and given that we don't have a specific need for a non-test caller to reset the state of the companion fields I worry that some future dev is going to - with good intentions - activate reset() when it's not necessary and thus reduce the value of the work in this PR .

Reply:
Renaming it `destroy` or `deinit` might address this ambiguity and reduce/eliminate the likelihood of misuse.

Retrieving the method by name and modifying its visibility in tests screams antipattern/smell to me (I don't feel strongly enough to get my crusading gear though).

There are situations where a `destroy` or `deinit` might be needed: in a previous PR this client used a bearer auth handler that would discard its token if a request resulted in a 403. Since the experience team might further maintain/refine this, we might consider packaging the functionality now (with an informative docstring even).

Reply:
ok I'll make the change

Reply:
@arnejduranovic the reason is it's called out by Oracle as "best practice" 

> Use the most restrictive access level that makes sense for a particular member. Use private unless you have a > good reason not to.

https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html

that being said y'all seem to want it public so I'll make it public

Reply:
@thetaurean [YAGNI](https://en.wikipedia.org/wiki/You_aren't_gonna_need_it) is why I was looking to keep it private. nowhere in the app is that method being called outside of test code and we can change it to public when we need to. 

that being said - y'all seem to want it public so I'll make it public.

Reply:
Good and valid points, David. I'm happy to approve the PR with either method, I think this is minor and don't think its a show-stopper. 

To me, you ARE using this method somewhere (in test code). The test code is the client calling code of this object and to me it makes sense for this method to be public because of that. I don't think we should add methods to classes just for testing purposes, but in this case I think a "reset" method makes sense for the public facing interface of this class given what it does.

Reply:
Actually, I want to edit and say I think we should not push code that is using reflection, even if its test code. So we should address this before merging.